src/components/Segmented/Segmented.tsx

```diff
@@ -0,0 +1,143 @@
+import { ReactElement, useState } from 'react'
+import classnames from 'classnames'
+
+import { Option, OptionsAlignments } from './Segmented.types'
+import { getOptionKey, isDisabledOption, isString, isVerticalOption, resolveKey } from './Segmented.utils'
+import styles from './Segmented.module.css'
+
+const { Horizontal, Vertical } = OptionsAlignments
+const { segmented, segmentedOption, disabled, selected, vertical } = styles
+
+export type SegmentedProps = {
+
+  /**
+   * The option initially selected. Either one of the following:
+   *
+   * - string: the key of the initially selected option
+   * - number: the position in the array of the initially selected option
+   */
+  defaultValue?: string | number
+
+  /**
+   * Determines whether the segmented control is disabled.
+   */
+  isDisabled?: boolean
+
+  /**
+   * A callback function that is invoked when the selected option changes.
+   * @param {Option} option - The newly selected option.
+   */
+  onChange?: (option: Option) => void
+
+ /**
+   * An array of segmented options to be displayed. Either one of the following:
+   *
+   * - string
+   * - object: {
+   *
+   *     icon?: ReactNode, The icon of the option
+   *
+   *     label: ReactNode, The selectable text of the option
+   *
+   *     isDisabled?: boolean, Whether the option is clickable
+   *
+   *     key: string, The value associated with the option
+   *
```

Comment:
```suggestion
   *     icon?: ReactNode - The icon of the option
   *     label: ReactNode - The selectable text of the option
   *     isDisabled?: boolean - Whether the option is clickable
   *     key: string - The value associated with the option
```

Reply:
This also comes witht a slightly less readable Storybook.

It would go from this
![image](https://github.com/mia-platform/design-system/assets/9254840/0c1d0585-6973-4771-a7f8-fb324bdc7651)

to this
![image](https://github.com/mia-platform/design-system/assets/9254840/318b0839-c044-4041-a407-f7aae2dda428)

to me it is not worth it.

Reply:
I didn’t think the visualization in the Storybook would be so drastically altered. Fair enough 👍🏼